src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -328,6 +344,10 @@ private void applySettings(ClusterChangedEvent event) {
             }
             String index = indexMetaData.index();
             IndexService indexService = indicesService.indexServiceSafe(index);
+            if (indexService == null) {
```

Comment:
in order to make this benefetial, you need to change the previous line not to use the Safe variant, since it throws an exception if the index is not there